### PR TITLE
Exit with 1 on error. Don't print a newline when ignore was used. Skip empty line and omit the comma

### PR DESCRIPTION
**Pull request by su8**
By default the program exits with 1 on several occasions, but doesn't exit with -1 on error. This is out to resolve @lasers wish in #121

### CODE REVIEW

File: prog/sensors/main.c [hunk 1]
```diff
@@ -361,20 +369,20 @@ int main(int argc, char *argv[])
...
 
 		if (!cnt) {
 			fprintf(stderr, "Specified sensor(s) not found!\n");
-			err = 1;
+			err_code = -1;
 		}
 	}
 
-exit:
+exitt:
```
[the remark this answers is not in the document]
Comment (su8) [reply prog/sensors/main.c]: The goto label shadows the `exit()` function :hankey:

[the remark this answers is not in the document]
Comment (su8) [reply prog/sensors/main.c]: `error_exit` is better name for this label.

[the remark this answers is not in the document]
Comment (lasers) [reply prog/sensors/main.c]: I like the third option. Fight with me irl. @cyrinux, can you test this when you can? Thx.

[the remark this answers is not in the document]
Comment (cyrinux) [reply prog/sensors/main.c]: Yes when I can ;)

**Comment by olysonek** [comment]
I haven't got around to read that issue yet, but exiting with a negative error code is very uncommon. It's possible even that it has some special meaning.

File: prog/sensors/main.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (lasers) [reply prog/sensors/main.c]: @cyrinux What error code do you get with `sensors -u`?

**Comment by su8** [comment]
Okay will rewrite it to exit with **1** instead **-1**, thanks for the explanation.

File: prog/sensors/chips.c [hunk 2]
```diff
@@ -105,12 +105,12 @@ void print_chip_json(const sensors_chip_name *name)
 					if (subCnt > 0)
 						printf(",\n");
 					printf("         \"%s\": %.3f", sub->name, val);
+					subCnt++;
 				}
 
 			} else {
 				printf("(%s)", label);
```
Comment (olysonek) [inline prog/sensors/chips.c]: What about this branch?

Comment (olysonek) [reply prog/sensors/chips.c]: No, I meant there should be subCnt++ on line 112 too.

Comment (su8) [reply prog/sensors/chips.c]: Added it :heart_eyes:

**Comment by olysonek** [comment]
Ok, thanks. I'll look at the 'Exit with 1 on error' commit later. But looking at it quickly, I don't like that it adds a global variable. There's quite enough of those, we don't need any more :). The functions should be changed to return an integer error code and that error code should propagate to main().

**Comment by cyrinux** [comment]
I tested this branch, I got exit 1 when my wifi is off.

**Comment by su8** [comment]
That's expected. Upon boot (right after your init system is loaded) the kernel probes your devices and loads the appropriate kernel modules for them and it creates the appropriate /proc and/or /sys directories. If the device is removable then remove it, reboot and test the program again, she should exit with 0 this time as the malfunction /proc and/or /sys entry won't exist. If the device is soldered then you have no choice, but to add it to **/etc/sensors.d/sensors.conf** as ignored and the program will exit with 0.

**Comment by su8** [comment]
That's impossible as once /proc and/or /sys entry is created it stays there until you reboot. That's what sensorsd parses and outputs, along with **sensors.conf** if presented.

File: prog/sensors/chips.c [hunk 3]
```diff
@@ -97,22 +100,26 @@ void print_chip_json(const sensors_chip_name *name)
...
-			subCnt++;
 		}
 		free(label);
 		printf("\n      }");
 		cnt++;
 	}
-	printf("\n");
+	if (0 != cnt) {
+		printf("\n");
+	}
```
Comment (lasers) [inline prog/sensors/chips.c]: @su8 Do we know for sure we need this?

**Comment by lasers** [comment]
@olysonek You may close this PR.

**Comment by lasers** [comment]
@olysonek I don't need it anymore. The solution was that I parse `sensors -u` into a Python dict instead of using `sensors -j` because initially, it might take a while for Debian and the likes to obtain `-j` support. 

Even if I don't need it anymore, the original issue should still be there. IIRC it was a comma that gets printed at each end of line. With a bad (or downed) sensor, it becomes a stray comma.

**Comment by olysonek** [comment]
Partially applied in 2c8cca3d6cd60.

**Comment by olysonek** [comment]
So a fix for the JSON problem has been applied. The only thing that remains is changing the exit code to 1 when getting a sensor reading fails. I have a WIP branch for that here:
https://github.com/olysonek/lm-sensors/tree/fix-exit-code

However there isn't currently any request for changing the exit code, so I'll keep it as it is for now, so that I don't break people's scripts unnecessarily.

**Pull request closed without merging**